### PR TITLE
Makes Contracts Slightly Less Godawful

### PR DESCRIPTION
**Pull request by Camryn-Buttes**
[BUG] [FEATURE][CLEANLINESS][QoL][REWORK]
## About the PR 

Fixes up a lot of the grodiness involved with chaplain contracts. Completely changes how the hair contract works, actually does proper tooltip updates for the weapon gear, adjusted some other contracts based on community feedback, some bug fixes, etc.

Also includes a template and guide for creating new contracts.

Long and short of it: code marginally less shitty

## Why's this needed? 

So I can sleep at night without being haunted by nightmares of buttescode past.

## Changelog

```changelog
(u)Camryn Buttes
(*)Revamped and rebalanced chaplain contracts. Replaced reversal contract with psychic meatcube contract.  Monkey soul farming is gone too. RIP 
```

### CODE REVIEW

File: code/WorkInProgress/infernal_contracts.dm
Comment (Camryn-Buttes) [reply code/WorkInProgress/infernal_contracts.dm:10]: I didn't even know that was a thing!

File: code/WorkInProgress/infernal_contracts.dm
Comment (Camryn-Buttes) [reply code/WorkInProgress/infernal_contracts.dm:17]: Intentional change, just forgot to make a comment

File: code/WorkInProgress/infernal_contracts.dm
Comment (Camryn-Buttes) [reply code/WorkInProgress/infernal_contracts.dm:294]: Not actually new code, but now that I look at it, yikes

File: code/datums/abilities/diabolical.dm
Comment (Camryn-Buttes) [reply code/datums/abilities/diabolical.dm:168]: SO THAT'S WHY IT WOULDN'T WORK

Comment (Camryn-Buttes) [reply code/datums/abilities/diabolical.dm:168]: It kept throwing an error about expecting a static value

File: code/WorkInProgress/infernal_contracts.dm
Comment (Camryn-Buttes) [reply code/WorkInProgress/infernal_contracts.dm:23]: whoops, forgot to set to 0 instead of null, thanks for catching

File: goonstation.dme
Comment (Camryn-Buttes) [reply goonstation.dme:28]: This was automatically added by VSC and I naively assumed that it was the correct place

**Comment by Camryn-Buttes** [comment]
Making a few notes for ease of use in the future + doing some additional testing, then I'll have one more commit

**Comment by Camryn-Buttes** [comment]
it is now, hopefully, done

**Comment by Camryn-Buttes** [comment]
I HAVE BEEN INFORMED THAT USING CONTRACTS ON MONKEYS STILL GIVES YOU SOULS.

I will be making an additional commit that rectifies this issue, please hold on review until this commit is made.

**Comment by Camryn-Buttes** [comment]
Aaaaaand there we go.

No more monkey soul farming.

**Comment by Camryn-Buttes** [comment]
There, that's it, I'm done, not touching it any more.

File: code/WorkInProgress/infernal_contracts.dm
Comment (ZeWaka) [inline code/WorkInProgress/infernal_contracts.dm:730]: not a big fan of krampuses in generally, they're so buggy

Comment (Camryn-Buttes) [reply code/WorkInProgress/infernal_contracts.dm:730]: They're not much different from a regular meatcube anymore, besides the lack of a popping timer and the artificially old age for the scream sound effects.

Unless you're referring to krampus 1.0 (the non-cube krampus), which this contract does **not** make use of.

**Comment by Camryn-Buttes** [comment]
I have some ideas for additional contracts, but I want yo wait until this get merged before working on them

**Review by Sovexe — APPROVED** [review]
im not sure about the krampus thing or related bugs so I'll let zewaka respond, otherwise code looks ok to at this point

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**
